source/dem/force_chains_visualization.cc

```diff
@@ -0,0 +1,511 @@
+ *
+ */
+
+#include <core/parameters_lagrangian.h>
+#include <core/tensors_and_points_dimension_manipulation.h>
+
+#include <dem/force_chains_visualization.h>
+#include <dem/particle_particle_contact_force.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_out.h>
+#include <deal.II/grid/tria.h>
+
+#include <deal.II/numerics/data_out.h>
+
+#include <cmath>
+#include <fstream>
+#include <iostream>
+
+using namespace DEM;
+
+
+
+template <
+  int                                                       dim,
+  Parameters::Lagrangian::ParticleParticleContactForceModel contact_model,
+  Parameters::Lagrangian::RollingResistanceMethod rolling_friction_model>
+ParticlesForceChains<dim, contact_model, rolling_friction_model>::
+  ParticlesForceChains(const DEMSolverParameters<dim> &dem_parameters_in)
+  : ParticleParticleContactForce<dim, contact_model, rolling_friction_model>(
+      dem_parameters_in)
+  , dem_parameters(dem_parameters_in)
+{
+  force_normal.push_back(0);
+  vertices.push_back(Point<3>(0, 0, 0));
+  vertices.push_back(Point<3>(0, 0, 0));
+}
+
+
+
+template <
+  int                                                       dim,
+  Parameters::Lagrangian::ParticleParticleContactForceModel contact_model,
+  Parameters::Lagrangian::RollingResistanceMethod rolling_friction_model>
+void
+ParticlesForceChains<dim, contact_model, rolling_friction_model>::
+  multi_general_cell(Triangulation<1, 3>         &tria,
+                     const std::vector<Point<3>> &vertices)
+{
+  const unsigned int       n_cells = vertices.size() / 2;
+  std::vector<CellData<1>> cells(n_cells, CellData<1>());
+  for (unsigned int i = 0; i < n_cells; ++i)
+    {
+      for (unsigned int j = 0; j < 2; ++j)
+        {
+          cells[i].vertices[j] = 2 * i + j;
+          cells[i].material_id = 0;
+        }
+    };
+  tria.create_triangulation(vertices, cells, SubCellData());
+}
+
+
+
+template <
+  int                                                       dim,
+  Parameters::Lagrangian::ParticleParticleContactForceModel contact_model,
+  Parameters::Lagrangian::RollingResistanceMethod rolling_friction_model>
+void
+ParticlesForceChains<dim, contact_model, rolling_friction_model>::
+  calculate_force_chains(DEMContactManager<dim> &container_manager,
+                         const double            dt)
+{
+  ParticleParticleContactForce<dim, contact_model, rolling_friction_model>
+    force_chains_object(dem_parameters);
+
+
+  auto &local_adjacent_particles = container_manager.local_adjacent_particles;
+  // auto &ghost_adjacent_particles =
+  // container_manager.ghost_adjacent_particles; auto
+  // &local_periodic_adjacent_particles =
+  //  container_manager.local_periodic_adjacent_particles;
+  // auto &ghost_periodic_adjacent_particles =
+  //  container_manager.ghost_periodic_adjacent_particles;
+  // auto &ghost_local_periodic_adjacent_particles =
+  //  container_manager.ghost_local_periodic_adjacent_particles;
```

Comment:
If this is not necessary, you can remove those :)

Reply:
I told her to keep those since she soon be working on the ghost cells... my fault

Reply:
No it's allright, we can keep it. Just add a comment asto why we are keeping this for now.